docs/using_custom_header.rst

```diff
@@ -0,0 +1,104 @@
+.. title:: Custom Header Component Documentation
+
+Custom Header Component
+=======================
+
+Overview
+--------
+
+The ``Header`` component is used to display a header with a provided ``mainMenuItems``,
+``secondaryMenuItems``, and ``userMenuItems`` props. If props are provided, the component will use them; otherwise,
+If any of the props ``(mainMenuItems, secondaryMenuItems, userMenuItems)`` are not provided, default
+items are displayed. This component provides flexibility in customization, making it suitable for a wide
+range of applications.
+
+Props Details
+-------------
+
+The `Header` component accepts the following **optional** props for customization:
+
+``mainMenuItems``
+*****************
+
+The main menu items is a list of menu items objects. On desktop screens, these items are displayed on the left, to the right of the logo icon and to the left of the secondary menu.
```

Comment:
```suggestion
The main menu items is a list of menu items objects. On desktop screens, these items are displayed on the left side next to the logo icon.
```

Original text was a little confusing to me without a visual. I don't think the secondary menu needs to be mentioned as the way this was initially written implies that there is no space in between the two menus

Reply:
makes sense